lib/windows.js

```diff
@@ -93,35 +136,47 @@ function windows() {
 	// The text as a JavaScript string
 	const windowTitle = wchar.toString(windowTextBufferClean);
 
-	// Allocate a buffer to store the process ID
-	const processIdBuffer = ref.alloc('uint32');
-	// Write the process ID creating the window to the buffer (it returns the thread ID, but it's not used here)
-	user32.GetWindowThreadProcessId(activeWindowHandle, processIdBuffer);
-	// Get the process ID as a number from the buffer
-	const processId = ref.get(processIdBuffer);
-	// Get a "handle" of the process
-	const processHandle = kernel32.OpenProcess(PROCESS_QUERY_LIMITED_INFORMATION, false, processId);
+	// Return the Process ID & Process Handle from the Active Window Handle
+	const [processId, processHandle] = getProcessIdAndHandle(activeWindowHandle);
 
 	if (ref.isNull(processHandle)) {
 		return undefined; // Failed to get process handle
 	}
 
-	// Set the path length to more than the Windows extended-length MAX_PATH length
-	const pathLengthBytes = 66000;
-	// Path length in "characters"
-	const pathLengthChars = Math.floor(pathLengthBytes / 2);
-	// Allocate a buffer to store the path of the process
-	const processFileNameBuffer = Buffer.alloc(pathLengthBytes);
-	// Create a buffer containing the allocated size for the path, as a buffer as it must be writable
-	const processFileNameSizeBuffer = ref.alloc('uint32', pathLengthChars);
-	// Write process file path to buffer
-	kernel32.QueryFullProcessImageNameW(processHandle, 0, processFileNameBuffer, processFileNameSizeBuffer);
-	// Remove null characters from buffer
-	const processFileNameBufferClean = ref.reinterpretUntilZeros(processFileNameBuffer, wchar.size);
-	// Get process file path as a string
-	const processPath = wchar.toString(processFileNameBufferClean);
-	// Get process file name from path
-	const processName = path.basename(processPath);
+	// Return the Process Path & Process Name from a Process Handle
+	let [processPath, processName] = getProcessPathAndName(processHandle);
+
+	let newProcessName;
+	let newProcessPath;
+
+	// ApplicationFrameHost & Universal Windows Platform Support
+	if (processName === 'ApplicationFrameHost.exe') {
+
+		const windowProcess = ffi.Callback('bool', ['pointer', 'int32'], (hwnd, lParam) => {
+
+			// Return the Process ID & Process Handle from the Active Window Handle
+			const [processId, processHandle] = getProcessIdAndHandle(hwnd);
+
+			if (ref.isNull(processHandle)) {
+				return false; // Failed to get process handle
+			}
+
+			let [processPathChild, processNameChild] = getProcessPathAndName(processHandle);
+
+			if (processNameChild !== processName) {
+				newProcessName = processNameChild;
+				newProcessPath = processPathChild;
```

Comment:
Shouldn't it stop iterating the child windows at this point?

Reply:
No, it calls this "callback function", synchronously for each child window. So each time that `EnumChildWindows` is called, this is evaluated n times per sub-windows.

Reply:
Leaving open for fedback.

Reply:
What I'm saying is that there's no point in continuing calling the callback function on child windows after we've found the one we're looking for, right? If so, we could return `false` here to stop the iteration.

Reply:
Ah, didn't realize returning false would end the callback, thanks for clarification.